Cesium.CodeGen/Ir/Expressions/Values/LValueInstanceField.cs

```diff
@@ -49,13 +79,53 @@ protected override FieldReference GetField(IEmitScope scope)
             return _field;
         }
 
+        List<FieldReference>? path = null;
+
         var valueTypeReference = _structType.Resolve(scope.Context);
         var valueTypeDef = valueTypeReference.Resolve();
 
-        var field = valueTypeDef.Fields.FirstOrDefault(f => f?.Name == _name)
-                ?? throw new CompilationException(
+        var field = valueTypeDef.Fields.FirstOrDefault(f => f?.Name == _name);
+
+        if (field == null)
+        {
+            path = new(1);
+
+            foreach (var f in valueTypeDef.Fields)
+                if (RecursiveBuildPath(_name, f, path))
+                {
+                    path.Add(new FieldReference(f.Name, f.FieldType, f.DeclaringType));
+                    return new UnionType.UnionFieldReference(path[0], path);
+                }
+
+            throw new CompilationException(
                     $"\"{valueTypeDef.Name.Replace("<typedef>", string.Empty)}\" has no member named \"{_name}\"");
+        }
+                
         _field = new FieldReference(field.Name, field.FieldType, field.DeclaringType);
         return _field;
+
+        static bool RecursiveBuildPath(string fieldName, FieldDefinition type, List<FieldReference> list)
+        {
+            var fieldType = type.FieldType.Resolve();
+            if (fieldType == null) return false;
+
+            foreach(var field in fieldType.Fields)
+            {
+                if (field.Name == fieldName)
+                {
+                    list.Add(new FieldReference(field.Name, field.FieldType, field.DeclaringType));
+                    return true;
+                }
+
+                var resolved = field.FieldType.Resolve();
+                if (resolved == null) return false;
+                if (resolved.Name.StartsWith("Union_") && RecursiveBuildPath(fieldName, field, list))
```

Comment:
Okay. I also wanted to make the name more synthetic like "Union_{UniqueNumbers}", but that caused tests with union to fail because each time there was a new name different from the last. Just rename it to "_Union_{types}"?

Reply:
For now I think just `_Union_{types}` should be enough. We may reconsider it later, but right now it's not a problem at all.